### ISSUE TITLE
Overlap between collection-method-value and collection-return 

### ISSUE THREAD

**Issue by smackfu** [wix/eslint-plugin-lodash#149]
We are using lodash/canonical and have a case where we are using `_.map` but should be using `_.foreach`, and that is triggering two different rules with the same advice:
```
[INFO]   64:9  error  Use value returned from _.map               lodash/collection-method-value
[INFO]   64:9  error  Do not use _.map without returning a value  lodash/collection-return
```
Our code:
```
        _.map(self.availableFieldsState, function (o) {
          o.selected = self.selectAll;
        });
```
Ideally rules would not overlap like this.

**Comment by ganimomer** [comment]
These rules aren't overlapping. There are actually two _different_ violations in this code sample.
- `collection-method-value` warns if you do not use the value *returned* from `_.map`. 
- `collection-return` warns if you do not *return* a value within `_.map`'s iteratee.

It looks like a fix for this code sample would be to use `_.forEach` instead of `map`, since you're not mapping every element of the collections to something, but executing an action for each one.

**Comment by smackfu** [comment]
@ganimomer Oh I see now.  Thanks!
